### PR TITLE
feat(REACTTODO-25): logic-create-tasks-group-feature-with-local-storage

### PR DESCRIPTION
**Pull request by macopul**

### CODE REVIEW

File: src/components/AddToDoSection/AddToDoSection.tsx [hunk 1]
```diff
@@ -3,32 +3,40 @@ import { useTaskStorage } from '../../hooks/useTaskStorage';
...
-  addTaskInputPlaceholder: 'Add New Task...',
+  addTaskInputPlaceholder: 'Add New Group or Task...',
 };
 
 const AddToDoSection = () => {
   // now we can use useTaskStorage in any place at any level - cause methods inside are operating on shared context state which is one for all children - look inside TaskStoreContextProvider.tsx and in App.tsx
-  const { addTaskToList } = useTaskStorage();
-  const [taskTitle, setTaskTitle] = useState('');
+  const { addTask, addGroupToList } = useTaskStorage();
+  const [Title, setTitle] = useState('');
```
Comment (woopul) [inline src/components/AddToDoSection/AddToDoSection.tsx]: variables lowercase - title, setTitle

File: src/components/AddToDoSection/AddToDoSection.tsx [hunk 2]
```diff
@@ -3,32 +3,40 @@ import { useTaskStorage } from '../../hooks/useTaskStorage';
...
-  const handleAddTaskButtonClick = () => {
-    if (!taskTitle) {
+  const handleAddButton = (action: string) => {
+    if (!Title) {
       window.alert('The task can not be empty');
       return;
     }
-    addTaskToList({ title: taskTitle, id: Math.random().toString(), checked: false });
-    setTaskTitle('');
+    switch (action) {
```
Comment (woopul) [inline src/components/AddToDoSection/AddToDoSection.tsx:19]: good idea with switch 👍

File: src/components/AddToDoSection/AddToDoSection.tsx [hunk 3]
```diff
@@ -3,32 +3,40 @@ import { useTaskStorage } from '../../hooks/useTaskStorage';
...
+      case 'task':
+        addTask({ title: Title, id: Math.random().toString(), checked: false });
+        break;
+      case 'group':
+        addGroupToList({ groupTitle: Title, groupId: Math.random().toString(), taskList: [] });
+        break;
```
Comment (woopul) [inline src/components/AddToDoSection/AddToDoSection.tsx:27]: missing default from switch, not sure if lint is configured that way or what

File: src/components/AddToDoSection/AddToDoSection.tsx [hunk 4]
```diff
@@ -3,32 +3,40 @@ import { useTaskStorage } from '../../hooks/useTaskStorage';
...
+    }
+    setTitle('');
   };
 
   return (
     <div className={styles.AddToDoSection}>
       <input
         type="text"
-        value={taskTitle}
+        value={Title}
```
Comment (woopul) [inline src/components/AddToDoSection/AddToDoSection.tsx]: lowercase

File: src/components/AddToDoSection/AddToDoSection.tsx [hunk 5]
```diff
@@ -3,32 +3,40 @@ import { useTaskStorage } from '../../hooks/useTaskStorage';
 import styles from './AddToDoSection.module.scss';
 
 const labels = {
-  addTaskInputPlaceholder: 'Add New Task...',
+  addTaskInputPlaceholder: 'Add New Group or Task...',
 };
 
 const AddToDoSection = () => {
```
Comment (woopul) [inline src/components/AddToDoSection/AddToDoSection.tsx:9]: f you add something please remember to style it at least a little, text is white again..
Adding max height to a button or style this to match input height, keep buttons on right side gruped together if multiple - is also not that much of work..
image

File: src/components/AddToDoInGroupSection/GroupAddToDoSection.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import { useState } from 'react';
+import { useTaskStorage } from '../../hooks/useTaskStorage';
+
+const labels = {
+  addTaskInputPlaceholder: 'Add New Task...',
+};
+
+type GroupAddToDoSectionType = {
```
Comment (woopul) [inline src/components/AddToDoInGroupSection/GroupAddToDoSection.tsx:9]: Is this neccessary? 
You could add condition inside prop i.e isInGroup, and then conditionally render only button with add task, rest logic would stay the same: 
image

File: src/hooks/useTaskStorage.tsx [hunk 7]
```diff
@@ -1,38 +1,113 @@
...
+    taskList,
+    localStorageTaskGroupStore,
+    setTaskGroupList,
+    taskGroupList,
+  } = useTaskStoreContext();
 
-  const addTaskToList = ({ title, id, checked }: TaskItemType) => {
-    localStorageTaskStore.set([...taskList, { title, id, checked }]);
-    const currentTasksList = localStorageTaskStore.get();
+  const addGroupToList = (group: TaskItemGroupType) => {
```
Comment (woopul) [inline src/hooks/useTaskStorage.tsx]: use just addTaskGroup

File: src/context/TastStoreContextProvider.tsx [hunk 8]
```diff
@@ -1,35 +1,64 @@
 /* eslint-disable @typescript-eslint/no-empty-function */
 import { createContext, useState } from 'react';
-import { TASKS_STORAGE_KEY } from '../constants/storage';
+import { TASKS_LIST_STORAGE_KEY, TASK_GROUPS_STORAGE_KEY } from '../constants/storage';
 import { createLocalStorageStore } from '../tools/sotrage';
 import { TaskItemType } from '../types/TaskItemType';
+import { TaskItemGroupType } from '../types/TaskItemGroupType';
 
 export type TaskStoreContextType = {
+  taskGroupList: TaskItemGroupType[];
```
Comment (woopul) [inline src/context/TastStoreContextProvider.tsx]: No, let's just change initial structure of the data - to be smthng like 
*added Array just for better readability
image

File: src/context/TastStoreContextProvider.tsx [hunk 9]
```diff
@@ -1,35 +1,64 @@
...
+  setTaskGroupList: React.Dispatch<React.SetStateAction<TaskItemGroupType[]>>;
+  localStorageTaskGroupStore: {
```
Comment (woopul) [inline src/context/TastStoreContextProvider.tsx]: let's keep one structure for taskStorage, do not create separate storage for task groups and taskList inside

File: src/context/TastStoreContextProvider.tsx [hunk 10]
```diff
@@ -1,35 +1,64 @@
...
 });
 
 export type TaskStoreProviderType = {
   children: React.ReactNode;
 };
 
 export function TaskStoreProvider({ children }: TaskStoreProviderType) {
-  const { set, get } = createLocalStorageStore<TaskItemType[]>();
-  const [taskList, setTaskList] = useState(get(TASKS_STORAGE_KEY) || []);
+  const { set: setListOfGroups, get: getListOfGroups } =
```
Comment (woopul) [inline src/context/TastStoreContextProvider.tsx]: keep provider clean, to be removed

File: src/context/TastStoreContextProvider.tsx [hunk 11]
```diff
@@ -1,35 +1,64 @@
...
+  };
+
+  const value = {
+    taskList,
+    setTaskList,
+    localStorageTaskListStore,
+    taskGroupList,
+    setTaskGroupList,
+    localStorageTaskGroupStore,
+  };
```
Comment (woopul) [inline src/context/TastStoreContextProvider.tsx:51]: To sum up - we want to get back to previous clean state - only one source of truth wit `const value = { taskList, setTaskList, localStorageTaskStore };`

taskList you can rename to taskStorage for sake of better naming -> taskStorage we want to contain upper name TaskStoreType

**Review by woopul — CHANGES_REQUESTED** [review]

File: src/components/AddToDoSection/AddToDoSection.tsx (hunk 1 above)
Comment (macopul) [reply src/components/AddToDoSection/AddToDoSection.tsx]: done

File: src/components/AddToDoSection/AddToDoSection.tsx (hunk 3 above)
Comment (macopul) [reply src/components/AddToDoSection/AddToDoSection.tsx:27]: I have added : 
default:
break;
Please let me know if it is correct

File: src/components/AddToDoSection/AddToDoSection.tsx (hunk 4 above)
Comment (macopul) [reply src/components/AddToDoSection/AddToDoSection.tsx]: done

File: src/components/AddToDoSection/AddToDoSection.tsx (hunk 5 above)
Comment (macopul) [reply src/components/AddToDoSection/AddToDoSection.tsx:9]: I checked this in Firefox as well as in Chrome, can You tell me if there is a way to test project in various browses ?
I can not see what You see as a bug

File: src/hooks/useTaskStorage.tsx (hunk 7 above)
Comment (macopul) [reply src/hooks/useTaskStorage.tsx]: done

File: src/context/TastStoreContextProvider.tsx (hunk 8 above)
Comment (macopul) [reply src/context/TastStoreContextProvider.tsx]: done as I showed You last time

File: src/context/TastStoreContextProvider.tsx (hunk 9 above)
Comment (macopul) [reply src/context/TastStoreContextProvider.tsx]: done with TaskStoreType, which contains groups and tasksList

File: src/context/TastStoreContextProvider.tsx (hunk 10 above)
Comment (macopul) [reply src/context/TastStoreContextProvider.tsx]: .

File: src/context/TastStoreContextProvider.tsx (hunk 11 above)
Comment (macopul) [reply src/context/TastStoreContextProvider.tsx:51]: Please check if naming is okay :)

File: src/components/Layout/Layout.module.scss [hunk 12]
```diff
@@ -1,9 +1,10 @@
 .LayoutComponent {
   width: 100vw;
   max-width: 1440px;
-  height: 100vh;
+  min-height: 100vh;
+  height: fit-content;
```
Comment (woopul) [inline src/components/Layout/Layout.module.scss:5]: redundant

File: src/context/TastStoreContextProvider.tsx [hunk 13]
```diff
@@ -1,35 +1,53 @@
...
 
 // set empty initial values of context
 export const TaskStoreContext = createContext<TaskStoreContextType>({
-  taskList: [],
-  setTaskList: () => {},
-  localStorageTaskStore: { get: () => [], set() {} },
+  taskStore: { groups: [], taskList: [] },
+  setTaskStore: () => {},
+  localStorageTaskStore: {
+    get: () => {
```
Comment (woopul) [inline src/context/TastStoreContextProvider.tsx:27]: `get: () => ({groups: [], taskList: []})`

File: src/hooks/useTaskStorage.tsx [hunk 14]
```diff
@@ -1,38 +1,107 @@
...
+  const addTaskGroup = (group: TaskItemGroupType) => {
+    localStorageTaskStore.set({ ...taskStore, groups: [...groups, group] });
+    const currentTaskStore = localStorageTaskStore.get();
+    setTaskStore(currentTaskStore);
+  };
+
+  const addTask = (task: TaskItemType) => {
+    if (task.groupId) {
+      const actualGroupList = groups.find((group) => group.groupId === task.groupId);
+      actualGroupList?.taskList.push(task);
```
Comment (woopul) [inline src/hooks/useTaskStorage.tsx:21]: if(!actualGroupList) { return}

**Review by woopul — APPROVED** [review]

File: src/components/AddToDoInGroupSection/GroupAddToDoSection.tsx (hunk 6 above)
Comment (woopul) [reply src/components/AddToDoInGroupSection/GroupAddToDoSection.tsx:9]: Please add to another PR

**Pull request merged**